DEV/org.openl.rules.util/src/org/openl/rules/util/Numbers.java

```diff
@@ -0,0 +1,84 @@
+     *          <tr><td>E</td><td>Number</td><td>E</td><td>Separates mantissa and exponent in scientific notation. Need not be quoted in prefix or suffix.</td></tr>
+     *          <tr><td>;</td><td>Subpattern boundary</td><td></td><td>Separates positive and negative subpatterns</td></tr>
+     *          <tr><td>%</td><td>Prefix or suffix</td><td>%</td><td>Multiply by 100 and show as percentage</td></tr>
+     *          <tr><td>‰</td><td>Prefix or suffix</td><td>‰</td><td>Multiply by 1000 and show as per mille value</td></tr>
+     *          <tr><td>'</td><td>Prefix or suffix</td><td></td><td>Used to quote special characters in a prefix or suffix, for example, "'#'#" formats 123 to "#123". To create a single quote itself, use two in a row: "# o''clock".</td></tr>
+     *     </tbody>
+     * </table>
+     * <p/>
+     * Some examples:
+     * <table border="1">
+     *     <thead>
+     *          <tr></th><th>Pattern&nbsp;</th><th>Number&nbsp;</th><th>Result&nbsp;</th><th>Comment&nbsp;</th></tr>
+     *     </thead>
+     *     <tbody>
+     *         <tr><td>0000&nbsp;</td><td>42&nbsp;</td><td>0042&nbsp;</td></tr>
+     *         <tr><td>0000.000&nbsp;</td><td>42.42&nbsp;</td><td>0042.420&nbsp;</td></tr>
+     *         <tr><td>0000.000&nbsp;</td><td>42&nbsp;</td><td>0042.000&nbsp;</td></tr>
+     *         <tr><td>#.##&nbsp;</td><td>3000.452&nbsp;</td><td>3000.45&nbsp;</td><td>&nbsp;</td></tr>
+     *         <tr><td>#.##&nbsp;</td><td>3000.4&nbsp;</td><td>3000.4&nbsp;</td><td>&nbsp;</td></tr>
+     *         <tr><td>#.##&nbsp;</td><td>3000&nbsp;</td><td>3000&nbsp;</td><td>&nbsp;</td></tr>
+     *         <tr><td>#,###.#&nbsp;</td><td>12000&nbsp;</td>12,000&nbsp;</td><td>&nbsp;</td></tr>
+     *         <tr><td>#0E0&nbsp;</td><td>12000&nbsp;</td>1.2E4&nbsp;</td><td>&nbsp;</td></tr>
+     *         <tr><td>003&nbsp;</td><td>&nbsp;</td>&nbsp;</td><td>&nbsp;</td></tr>
+     *         <tr><td>000;neg 0&nbsp;</td><td>3&nbsp;</td>003&nbsp;</td><td>&nbsp;</td></tr>
+     *         <tr><td>000;neg 0&nbsp;</td><td>-3&nbsp;</td>neg 003&nbsp;</td><td>For negative numbers positive pattern enhanced by negative pattern is used&nbsp;</td></tr>
+     *         <tr><td>0%&nbsp;</td><td>0.03&nbsp;</td>3%&nbsp;</td><td>&nbsp;</td></tr>
+     *         <tr><td>0‰&nbsp;</td><td>0.03&nbsp;</td>30‰&nbsp;</td><td>&nbsp;</td></tr>
+     *         <tr><td>'#'0'%'&nbsp;</td><td>3&nbsp;</td>#3%&nbsp;</td><td>&nbsp;</td></tr>
+     *     </tbody>
+     * </table>
+     *
+     *
+     * @param number any instance of {@link Number}
+     * @param pattern target pattern
+     * @return string representation of a given number.
+     */
+    public static String toString(Number number, String pattern) {
+        if (number == null || pattern == null) {
+            return null;
+        }
+        NumberFormat decimalFormat = new DecimalFormat(pattern, DecimalFormatSymbols.getInstance(Locale.US));
+        return decimalFormat.format(number);
```

Comment:
Process all exception which can be thrown while formatting, for example invalid pattern or number.

Reply:
It doesn't seem that this method throws ArithmeticException, but I covered all other possible exceptions and added then into the tests.